content/blog/2023-11-23-announcing-axum-0-7-0.md

```diff
@@ -0,0 +1,184 @@
+---
+date: "2023-11-23"
+title: "Announcing axum 0.7.0"
+description: "November 23, 2022"
+---
+
+Today, we're happy to announce [`axum`] version 0.7. `axum` is an ergonomic
+and modular web framework built with [`tokio`], [`tower`], and [`hyper`].
+
+This also includes new major versions of [`axum-core`], [`axum-extra`], and
+[`axum-macros`].
+
+## `hyper` 1.0 support
+
+The headline feature of `axum` 0.7 is support for `hyper` 1.0. `hyper` is a
+foundational library for much of the networking ecosystem in Rust and finally
+having a stable API is a big milestone.
+
+`hyper` 1.0 comes with a big shuffling of the APIs. The previous low level APIs
+(found in [`hyper::server::conn`]) is what has been stabilized whereas the high
+level APIs (such as [`hyper::Server`]) has been removed.
+
+The plan is to add the high level APIs to a new crate called [`hyper-util`].
+There we can build out the APIs without worrying too much about stability
+guarantees and backwards compatibility. When something is ready for
+stabilization it can be moved into `hyper`.
+
+`hyper-util` is still in early stages of development and some things (like the
+previous `Server` type) are still missing.
+
+Because `hyper-util` is not stable we don't want it to be part of `axum`'s
+public API. If you want to use something from `hyper-util` you have to depend
+it directly.
+
+`axum` 0.7 also requires [`tower-http`] 0.5.
+
+## A new `axum::serve` function
+
+`axum` 0.6 provided `axum::Server` which was an easy way to get started.
+`axum::Server` was however just a re-export of `hyper::Server` which has been
+removed from `hyper` 1.0.
+
+There isn't yet a full replacement in `hyper-util` so `axum` now provides its
+own `serve` function:
+
+```rust
+use axum::{
+    routing::get,
+    Router,
+};
+use tokio::net::TcpListener;
+
+let app = Router::new().route("/", get(|| async { "Hello, World!" }));
+
+let listener = TcpListener::bind("0.0.0.0:3000").await.unwrap();
+axum::serve(listener, app).await.unwrap();
+```
+
+The purpose of `axum::serve` is provide a way to get started with `axum`
+quickly and as such it does not support any configuration options whatsoever.
+If you need configuration you have to use `hyper` and `hyper-util` directly. We
+provide an example showing how to do that [here][hyper-serve-example].
```

Comment:
Should we maybe merge the graceful shutdown example (really hyper's one, since the axum one hasn't been updated yet) into that one? I think graceful shutdown is probably the most common hyper functionality people used that's not available with `axum::serve`.

Reply:
Good idea! I've filed an issue for it https://github.com/tokio-rs/axum/issues/2352